game/scripts/vscripts/components/duels/duels.lua

```diff
@@ -0,0 +1,309 @@
+    players = {
+    }
+  })
+
+  Duels.zone2 = ZoneControl:CreateZone('duel_2', {
+    mode = ZONE_CONTROL_INCLUSIVE,
+    players = {
+    }
+  })
+
+  GameEvents:OnHeroKilled(function (keys)
+    Duels:CheckDuelStatus(keys)
+  end)
+
+  Timers:CreateTimer(5, function ()
+    Duels:StartDuel()
+  end)
+end
+
+function Duels:CheckDuelStatus (keys)
+  if not Duels.currentDuel then
+    return
+  end
+
+  local playerId = keys.killed:GetPlayerOwnerID()
+  local foundIt = false
+
+  Duels:AllPlayers(Duels.currentDuel, function (player)
+    if foundIt or player.id ~= playerId then
+      return
+    end
+    foundIt = true
+    local scoreIndex = player.team .. 'Living' .. player.duelNumber
+    DebugPrint('Found dead player on ' .. player.team .. ' team with scoreindex ' .. scoreIndex)
+
+    Duels.currentDuel[scoreIndex] = Duels.currentDuel[scoreIndex] - 1
+
+    if Duels.currentDuel[scoreIndex] <= 0 then
+      Duels.currentDuel['duelEnd' .. player.duelNumber] = true
+      DebugPrint('Duel number ' .. scoreIndex .. ' is over and ' .. player.team .. ' lost')
+    end
+
+    if Duels.currentDuel.duelEnd1 and Duels.currentDuel.duelEnd2 then
+      DebugPrint('both duels are over, resuming normal play!')
+      Duels:EndDuel()
+    end
+  end)
+end
+
+function Duels:StartDuel ()
+  if Duels.currentDuel then
+    DebugPrint ('There is already a duel running')
+  end
+  -- respawn everyone
+  local goodPlayerIndex = 1
+  local badPlayerIndex = 1
+
+  local goodPlayers = {}
+  local badPlayers = {}
+
+  for playerId = 0,19 do
+    local player = PlayerResource:GetPlayer(playerId)
+    if player ~= nil then
+      DebugPrint ('Players team ' .. player:GetTeam())
+      if player:GetTeam() == 3 then
+        badPlayers[badPlayerIndex] = Duels:SavePlayerState(player:GetAssignedHero())
+        badPlayers[badPlayerIndex].id = playerId
+        badPlayers[badPlayerIndex].team = 'bad'
+        badPlayerIndex = badPlayerIndex + 1
+
+      elseif player:GetTeam() == 2 then
+        goodPlayers[goodPlayerIndex] = Duels:SavePlayerState(player:GetAssignedHero())
+        goodPlayers[goodPlayerIndex].id = playerId
+        goodPlayers[goodPlayerIndex].team = 'good'
+        goodPlayerIndex = goodPlayerIndex + 1
+
+      end
+
+      Duels:ResetPlayerState(player:GetAssignedHero())
+      -- disable respawn
+      player:GetAssignedHero():SetRespawnsDisabled(true)
+    end
+  end
+
+  goodPlayerIndex = goodPlayerIndex - 1
+  badPlayerIndex = badPlayerIndex - 1
+
+  -- split up players, put them in the duels
+  local maxPlayers = math.min(goodPlayerIndex, badPlayerIndex)
+
+  DebugPrint('Max players per team for this duel ' .. maxPlayers)
+
+  if maxPlayers < 1 then
+    DebugPrint('There aren\'t enough players to start the duel')
+    ShowMessage('There aren\'t enough players to start the duel')
+    return
+  end
+
+  -- local playerSplitOffset = math.random(1, maxPlayers)
+  local playerSplitOffset = maxPlayers
+  local spawnLocations = math.random(0, 1) == 1
```

Comment:
i think that's true in lua as well. no reason to be implicit when explicit is free